tests/test_dictdiffer.py

```diff
@@ -195,6 +195,11 @@ def test_types(self):
         diffed = next(diff(first, second))
         assert ('change', 'a', (['a'], 'a')) == diffed
 
+    def test_nan(self):
+        value = float('nan')
+        diffed = list(diff([value], [value]))
+        assert diffed == []
+    
```

Comment:
I was more thinking about `diff([float('nan')], [3.5]))` just to have it covered.